app/user/userController.js

```diff
@@ -0,0 +1,84 @@
+import crypto from 'crypto'
+import passport from 'koa-passport'
+import LocalStrategy from 'passport-local'
+import User from './userModel'
+
+
+const authenticate = (type,ctx) =>{
+                                    return passport.authenticate(type,(err, user, info, status) => {
+                                        if (user) {
+                                            ctx.status = 200
+                                            ctx.body = {
+                                                data:{  
+                                                        email:user.email,
+                                                        id:user._id,
+                                                        name:user.name,
+                                                        type:user.type
+                                                    }
+                                            }
+                                            return ctx.login(user)
+                                        } else {
+                                            ctx.status = 400
+                                            ctx.body = { 
+                                                status: 'error',
+                                                data:info
+                                                }
+                                            }
+                                    })(ctx)
+                                }
+
+const response = (user,ctx) =>{
+                    if(user){
+                        ctx.status = 200
+                        ctx.body={
+                            data:{  
+                                email:user.email,
+                                id:user._id,
+                                name:user.name,
+                                type:user.type
+                            }
+                        }
+                    }else if(!user){
+                        ctx.status = 200
```

Comment:
this should be ctx.status = 400, since you're returning an error.
Also make your error messages consistent, up above you did this 
```
ctx.body = { 
 status: 'error',
 data:info
}
```
And down here, you're returning just data.
Instead make it return `data: {message: 'error message'}`